src/Sylius/Component/Payment/spec/Resolver/CompositeMethodsResolverSpec.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Payment\Resolver;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Sylius\Component\Payment\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Resolver\MethodsResolverInterface;
+use Sylius\Component\Registry\PrioritizedServiceRegistryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class CompositeMethodsResolverSpec extends ObjectBehavior
+{
+    function let(PrioritizedServiceRegistryInterface $resolversRegistry)
+    {
+        $this->beConstructedWith($resolversRegistry);
+    }
+    
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Payment\Resolver\CompositeMethodsResolver');
+    }
+
+    function it_implements_Sylius_payment_methods_resolver_interface()
+    {
+        $this->shouldImplement(MethodsResolverInterface::class);
+    }
+
+    function it_uses_registry_to_provide_payment_methods_for_payment_subject(
```

Comment:
``for_payment``? It is ``PaymentInterface`` not ``PaymentSubjectInterface`` used.